clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10594,39 +10587,33 @@ static std::string getSYCLPostLinkOptimizationLevel(const ArgList &Args) {
   return "-O2";
 }
 
-// sycl-post-link tool normally outputs a file table (see the tool sources for
-// format description) which lists all the other output files associated with
-// the device LLVMIR bitcode. This is basically a triple of bitcode, symbols
-// and specialization constant files. Single LLVM IR output can be generated as
-// well under an option.
-//
-void SYCLPostLink::ConstructJob(Compilation &C, const JobAction &JA,
-                             const InputInfo &Output,
-                             const InputInfoList &Inputs,
-                             const llvm::opt::ArgList &TCArgs,
-                             const char *LinkingOutput) const {
-  const SYCLPostLinkJobAction *SYCLPostLink =
-      dyn_cast<SYCLPostLinkJobAction>(&JA);
-  // Construct sycl-post-link command.
-  assert(SYCLPostLink && "Expecting SYCL post link job!");
-  ArgStringList CmdArgs;
+static void addArgs(ArgStringList &DstArgs, const llvm::opt::ArgList &Alloc,
+                    ArrayRef<StringRef> SrcArgs) {
+  for (const auto Arg : SrcArgs) {
+    DstArgs.push_back(Alloc.MakeArgString(Arg));
+  }
+}
 
-  llvm::Triple T = getToolChain().getTriple();
+static void getOtherSYCLPostLinkOpts(const ToolChain &TC, const JobAction &JA,
+                                     const llvm::opt::ArgList &TCArgs,
+                                     llvm::Triple Triple,
+                                     ArgStringList &PostLinkArgs,
+                                     bool SpecConsts, types::ID OutputType) {
   // See if device code splitting is requested
   if (Arg *A = TCArgs.getLastArg(options::OPT_fsycl_device_code_split_EQ)) {
     auto CodeSplitValue = StringRef(A->getValue());
     if (CodeSplitValue == "per_kernel")
-      addArgs(CmdArgs, TCArgs, {"-split=kernel"});
+      addArgs(PostLinkArgs, TCArgs, {"-split=kernel"});
     else if (CodeSplitValue == "per_source")
-      addArgs(CmdArgs, TCArgs, {"-split=source"});
+      addArgs(PostLinkArgs, TCArgs, {"-split=source"});
     else if (CodeSplitValue == "auto")
-      addArgs(CmdArgs, TCArgs, {"-split=auto"});
+      addArgs(PostLinkArgs, TCArgs, {"-split=auto"});
     else { // Device code split is off
     }
-  } else if (T.getArchName() != "spir64_fpga") {
+  } else if (Triple.getArchName() != "spir64_fpga") {
```

Comment:
I think we can try to move all 'Triple' related options to a separate function getTripleRelatedSYCLPostLinkOpts(....) and call it selectively for the old offload model.

Reply:
New function created to move triple related items into.